### DISCUSSION TITLE
Tree height listing and export

### DISCUSSION THREAD

**Discussion by mloricn2** [r-lidar/lidR#741] [Q&A] [answered]
Dear Author!

My question would be, how can it be solved so that not only a dot is placed on the top of the detected trees, but their height (Z)?

```
x <- plot(nlas, color = "RGB", bg = "white", size = 4)
add_treetops3d(x, ttops)
```
In addition, I would like to export the LAS file in some 3D format so that it is connected with the tree height points.

Thanks for the help!

**Comment by Jean-Romain** [comment, marked as answer]
You can use `rgl::texts3d`

```r
library(lidR)
LASfile <- system.file("extdata", "MixedConifer.laz", package="lidR")
las <- readLAS(LASfile, select = "xyzc", filter = "-inside_circle 481300 3812980 25")

dtm <- rasterize_terrain(las, algorithm = tin())
ttops <- locate_trees(las, lmf(ws = 5))

xy = sf::st_coordinates(ttops)

x <- plot(las)
add_treetops3d(x, ttops)
rgl::texts3d(xy[,1]-x[1], xy[,2]-x[2], xy[,3]+1.5, round(xy[,3],0), col = "white")
```

**Reply by mloricn2** [reply]
Thank you very much, this solved my problem!

Is it possible to export the point cloud so that the ttops points and the height are also exported?

Thank you!

**Reply by Jean-Romain** [reply]
I do not understand the question. The las object contains all the points, the sf object contains the coordinates of all the tree tops found. You can write the sf object in a geospatial file such as a geopackage
